crates/pallet-feeds/src/lib.rs

```diff
@@ -29,13 +35,32 @@ mod mock;
 #[cfg(all(feature = "std", test))]
 mod tests;
 
+/// Simple index type that we use to count feeds.
+pub type FeedId = u64;
+
+/// Arbitrary user data stored to the network.
+///
+/// The size of each object is typically limited by the transaction size limit.
+pub type Object = Vec<u8>;
+
+/// Some meta information about [`Object`].
+pub type Metaobject = Vec<u8>;
```

Comment:
I don't think these need to be public. Also why are they moved out if they are not used anywhere except pallet module at all?

Reply:
`Metaobject` is incorrect, it is not a metaobject, it is a piece of metadata that it associated with some object.
It is `Metadata`, but since `Metadata` is extremely generic, `ObjectMetadata` was used before and let's keep it that way.

Reply:
>I don't think these need to be public. Also why are they moved out if they are not used anywhere except pallet module at all?

They are well documented and will be exposed in rustdoc that will help the reader understand more. I agree they don't have to be public for now, but if doing that can be beneficial, why not? 

>Metaobject is incorrect, it is not a metaobject

Maybe not 100% precise, but if we agree with the Object terminology, don't see why we can't borrow this concept here. According to https://en.wikipedia.org/wiki/Metaobject, _In computer science, a metaobject is an object that manipulates, creates, describes, or implements objects (including itself)._, so a metaobject can be used to describe objects, then it plays a role of Metadata to some extent.

But no strong opinion on this, we can use `Metadata` or `ObjectMetadata`, just disagree it's incorrect.

Reply:
It doesn't have to be in docs. In fact, right now type aliases are de-sugared when rendering with rustdoc: https://subspace.github.io/subspace/pallet_feeds/enum.Call.html
They are just convenience type aliases, with the same success they may not have been there at all.

Reply:
I take my word on incorrectness back, but as current naming is already in drafts, specs, diagrams and discussions, the question becomes why `Metaobject` all of a sudden. These are the kinds of things that should have a solid reason to be renamed into something we didn't use before and discussed first. I just don't understand why it was necessary.

Reply:
That's fair if we're considering the burden of history and migration. We don't have to do this, of course, just wanted to make it look nicer.

Reply:
That's fair if we're considering the burden of history and migration. We don't have to do this, of course, just wanted to make it look nicer.

Reply:
We choose `Object` and `ObjectMetadata` eventually.